core/trino-main/src/main/java/io/trino/sql/rewrite/ShowQueriesRewrite.java

```diff
@@ -297,7 +297,9 @@ protected Node visitShowRoles(ShowRoles node, Void context)
 
             if (node.isCurrent()) {
                 accessControl.checkCanShowCurrentRoles(session.toSecurityContext(), catalog);
-                List<Expression> rows = metadata.listEnabledRoles(session, catalog.orElseThrow()).stream()
+                Set<String> enabledRoles = catalog.map(c -> metadata.listEnabledRoles(session, c))
+                        .orElseGet(() -> session.getIdentity().getEnabledRoles());
```

Comment:
Using `session.getIdentity().getEnabledRoles()` looks incorrect as they are system roles, not catalogs roles which expect here.

Reply:
The catalog check should have been removed.  If the catalog is present, you get the catalog roles, otherwise you get the system roles